flink-connector/src/main/resources/META-INF/services/org.apache.flink.table.factories.Factory

```diff
@@ -0,0 +1,7 @@
+#
+# Copyright 2024 Datastrato Pvt Ltd.
+# This software is licensed under the Apache License version 2.
+#
+
+com.datastrato.gravitino.flink.connector.store.GravitinoCatalogStoreFactory
+com.datastrato.gravitino.flink.connector.hive.GravitinoHiveCatalogFactory
```

Comment:
If supporting Iceberg, just provide another `GravitinoIcebergCatalogFactory` ?

Reply:
@FANNG1 Yes. This is what I want to talk about. Do we implement one CatalogFactory for each storage, or just on GravitinoCatalogFactory? If we use one GravitinoCatalogFactory, we may need another property to identify which real catalog should be used. 

Another question is do we need to use GravitinoCatalog for all storage. I think the CatalogFactory should be consistent with the Catalog. Flink supports registering the catalog manually, if we only give a GravitinoCatalog, we can also simplify the usage.
 
The original usage:https://nightlies.apache.org/flink/flink-docs-release-1.18/docs/dev/table/catalogs/#using-sql-ddl
```
// Create a HiveCatalog 
Catalog catalog = new HiveCatalog("myhive", null, "<path_of_hive_conf>");

// Register the catalog
tableEnv.registerCatalog("myhive", catalog);
```

Use one GravitinoCatalog.
```
Map<String, String> properties = new HashMap<>();
**properties.put("catalog.type", "hive"); // another property used to identify the real catalog**

Catalog catalog = new GravitinoCatalog("gravitino", properties)

// Register the catalog
tableEnv.registerCatalog("myhive", catalog);
```

Use different GravitinoCatalog for different storage:
```
Catalog catalog = new GravitinoHiveCatalog("gravitino", xxx)

// Register the catalog
tableEnv.registerCatalog("myhive", catalog);
```

which do you prefer? Can you share your thoughts? Thanks.

Reply:
Can you also help review it? Thanks. @hackergin

Reply:
I prefer different GravitinoCatalog, because this solution seems simple for user.

Reply:
From my understanding, the Catalog Store is essentially for storing the configuration of Flink Catalog. Therefore, if users create a catalog using the following DDL, we should support saving the configuration of the Catalog.  

```SQL
CREATE CATALOG myhive WITH (
  'type' = 'hive',
  'hive-conf-dir' = '/opt/hive-conf'
);
```

At the same time, when constructing a real Catalog instance, we can automatically create a GravitinoCatalog instead of a HiveCatalog.

And we should prohibit the direct creation of Catalog instances.
```java
tableEnv.registerCatalog("myhive", catalog);
```
When using the Catalog instance to register this Catalog with the table env, this Catalog will not be persisted to the Catalog Store.